src/di/dependency_injection.ts

```diff
@@ -0,0 +1,64 @@
+import { ChaptersController } from "../controllers/chapters_controller";
+import { ExercisesController } from "../controllers/exercises_controller";
+import { TopicsController } from "../controllers/topics_controller";
+import { PostgresChapterDAO } from "../dao/impl/postgres/postgres_chapters_dao";
+import { PostgresExerciseDAO } from "../dao/impl/postgres/postgres_exercises_dao";
+import { PostgresTopicsDAO } from "../dao/impl/postgres/postgres_topics_dao";
+import { ChaptersService } from "../services/impl/chapters_service.impl";
+import { ExercisesService } from "../services/impl/exercises_service.impl";
+import { TopicsService } from "../services/impl/topics_service.impl";
+
+// For now, we create objects "manually".
+// TODO: use a specialized container
+
+let topics_controller: TopicsController | null = null;
+let chapters_controller: ChaptersController | null = null;
+let exercises_controller: ExercisesController | null = null;
+
+export function createTopicsController(): TopicsController {
+     if (topics_controller !== null) {
```

Comment:
When will this be true ?
topics_controller is not exported, thus it will always be null in this function

Reply:
if you call this function 2 times, you would get 2 different TopicsController without this check. Creating a controller might become expensive as we need to create a DAO implementation and a service.

Reply:
if you call this function a second time, will the topics_controller be the same in the if statement as the first one (in the 1st call)

Reply:
I'm not sure we should check